iosApp/iosApp/Pages/RouteDetails/CollapsableStopList.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  CollapsableStopList.swift
+//  iosApp
+//
+//  Created by Melody Horn on 7/8/25.
+//  Copyright © 2025 MBTA. All rights reserved.
+//
+
+import Shared
+import SwiftUI
+
+struct CollapsableStopList<RightSideContent: View>: View {
+    let lineOrRoute: RouteCardData.LineOrRoute
+    let segment: RouteDetailsStopList.Segment
+    let onClick: (RouteDetailsStopList.Entry) -> Void
+    let isFirstSegment: Bool
+    let isLastSegment: Bool
+    let rightSideContent: (RouteDetailsStopList.Entry) -> RightSideContent
+
+    @State var stopsExpanded = false
+
+    let inspection = Inspection<Self>()
+
+    init(
+        lineOrRoute: RouteCardData.LineOrRoute,
+        segment: RouteDetailsStopList.Segment,
+        onClick: @escaping (RouteDetailsStopList.Entry) -> Void,
+        isFirstSegment: Bool = false,
+        isLastSegment: Bool = false,
+        rightSideContent: @escaping (RouteDetailsStopList.Entry) -> RightSideContent
+    ) {
+        self.lineOrRoute = lineOrRoute
+        self.segment = segment
+        self.onClick = onClick
+        self.isFirstSegment = isFirstSegment
+        self.isLastSegment = isLastSegment
+        self.rightSideContent = rightSideContent
+    }
+
+    var body: some View {
+        if let stop = segment.stops.first, segment.stops.count == 1 {
+            StopListRow(
+                stop: stop.stop,
+                onClick: { onClick(stop) },
+                routeAccents: .init(route: lineOrRoute.sortRoute),
+                stopListContext: .routeDetails,
+                connectingRoutes: stop.connectingRoutes,
+                stopPlacement: .init(isFirst: isFirstSegment, isLast: isLastSegment, includeLineDiagram: false),
+                descriptor: { Text("Less common stop").font(Typography.footnote) },
+                rightSideContent: { rightSideContent(stop) }
+            ).background(Color.fill1)
+                .onReceive(inspection.notice) { inspection.visit(self, $0) }
+        } else {
+            DisclosureGroup(isExpanded: $stopsExpanded, content: {
```

Comment:
thought(non-blocking) Nice that DisclosureGroup gives the expand/collapse voice over hint by default, which we had to manually add in android!